### PR TITLE
Add Gong Oauth provider core

### PR DESCRIPTION
**Pull request by albandum**
## Description

Add Gong as an oAuth provider in core
Gong documentation : https://help.gong.io/docs/create-an-app-for-gong#set-up-oauth-access

### CODE REVIEW

**Comment by spolu** [comment]
@albandum can you link to the Gong documentation in the PR description as well as https://www.notion.so/dust-tt/OAuth-service-remove-Nango-72fdcee1b51a49d3bcb4dd5a2fb93bf0?pvs=4 :pray:

**Comment by albandum** [comment]
> @albandum can you link to the Gong documentation in the PR description as well as https://www.notion.so/dust-tt/OAuth-service-remove-Nango-72fdcee1b51a49d3bcb4dd5a2fb93bf0?pvs=4 🙏

Done!

File: core/src/oauth/providers/gong.rs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            },
+            _ => Err(anyhow!("Missing `expires_in` in response from Gong"))?,
+        };
+
+        let refresh_token = match raw_json["refresh_token"].as_str() {
+            Some(token) => token,
+            None => Err(anyhow!("Missing `refresh_token` in response from Gong"))?,
+        };
+
+        Ok(FinalizeResult {
```
Comment (flvndvd) [inline core/src/oauth/providers/gong.rs:85]: From my understanding of their documentation, we also need to save `api_base_url_for_customer`. All queries should redirected to this url, do you confirm?

File: core/src/oauth/providers/gong.rs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    }
+
+    async fn refresh(&self, connection: &Connection) -> Result<RefreshResult, ProviderError> {
+        let refresh_token = match connection.unseal_refresh_token() {
+            Ok(Some(token)) => token,
+            Ok(None) => Err(anyhow!("Missing `refresh_token` in Gong connection"))?,
+            Err(e) => Err(e)?,
+        };
+
+        let body = json!({
```
Comment (flvndvd) [inline core/src/oauth/providers/gong.rs]: Based on the documentation, this should be done using query parameters no body and only `grant_type` and `refresh_token` are required. The Authorization header should be set like you do in `finalize`.

**Review by flvndvd — CHANGES_REQUESTED** [review]

File: core/src/oauth/providers/gong.rs (hunk 1 above)
Comment (albandum) [reply core/src/oauth/providers/gong.rs:85]: Looks like it - missed this part 😓 
Will take another pass

**Review by flvndvd — APPROVED** [review]
LGTM

**Pull request merged**